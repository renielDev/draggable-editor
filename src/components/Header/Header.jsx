import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

const HeaderStyled = ({ tag, children, ...rest }) => {
  const Container = styled(tag)`
    color: red;
  `

  return <Container {...rest}>{children}</Container>
}

HeaderStyled.propTypes = {
  children: PropTypes.node,
  tag: PropTypes.string,
}

HeaderStyled.defaultProps = {
  children: '',
  tag: 'h1',
}

const Header = props => {
  return (
    <HeaderStyled {...props}>
      <span>Sample deyms</span>
    </HeaderStyled>
  )
}

export default Header
