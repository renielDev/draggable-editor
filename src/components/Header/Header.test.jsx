import React from 'react'
import { render } from '@testing-library/react'
import Header from './Header'

describe('<Header />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(<Header />)
      expect(container).toMatchSnapshot()
    })
  })
})
