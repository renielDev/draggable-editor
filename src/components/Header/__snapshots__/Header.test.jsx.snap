// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Snapshots should match snapshot 1`] = `
<div>
  <h1>
    Sample header
  </h1>
</div>
`;
